Snakefile

```diff
@@ -0,0 +1,94 @@
+    """
+    input:
+        metadata = 'inputs/metadata-paired-mgx-mtx.tsv',
+        sigs = expand('outputs/sourmash_sketch/{run_accession}.sig', run_accession = RUN_ACCESSIONS) 
+    output:
+        sigs = expand("outputs/sourmash_sketch_subtract/{sample}_k{{ksize}}.sig", sample = SAMPLES)
+    run:
+        ksize = wildcards.ksize
+        # read in metadata dataframe to derive sample pairs
+        metadata = pd.read_csv(input.metadata, sep = "\t") # read in metadata as pandas df
+        metadata = metadata.reset_index()  # make sure indexes pair with number of rows
+        
+        for index, row in metadata.iterrows():
+            # grab the run accessions for a given paired metagenome and metatranscriptome
+            mtx_run_accession = row['mtx_run_accession']
+            mgx_run_accession = row['mgx_run_accession']
+            # using the run assession, create paths of the signatures
+            mtx_sigfile = os.path.join('outputs/sourmash_sketch', mtx_run_accession + '.sig')
+            mgx_sigfile = os.path.join('outputs/sourmash_sketch', mgx_run_accession + '.sig')
+
+            # read in mtx signature and grab both the minhash object and the hashes in that object
+            mtx_sigobj = sourmash.load_one_signature(mtx_sigfile, ksize=ksize, select_moltype='DNA')
+            mtx_mh = from_sigobj.minhash
+            # turn the mtx hashes into a set
+            mtx_subtract_mins = set(mtx_mh.hashes)
+
+            # read in mgx signature
+            mgx_sigobj = sourmash.load_one_signature(mgx_sig_path, ksize=ksize, select_moltype='DNA')
+
+            # do the subtraction
+            mtx_subtract_mins -= set(mgx_sigobj.minhash.hashes)
+
+            # make a new signature that only contains the mtx hashes that were not in the mgx
+            mtx_subtract_mh = mtx_sigobj.minhash.copy_and_clear().flatten()
+            mtx_subtract_mh.add_many(mtx_subtract_mins)
+
+            # re-establish hash abundances from mtx sig
+            # re-read in mtx sig just in case abundances were removed in place
+            abund_sig = sourmash.load_one_signature(mtx_sig_path, ksize=ksize, select_moltype='DNA')
+            mtx_subtract_mh = mtx_subtract_mh.inflate(abund_sig.minhash)
+  
+            # create a new sig object that can be written to file
+            mtx_subtract_sigobj = sourmash.SourmashSignature(mtx_subtract_mh)
+
+            # create a name that reflects the signature contents
+            # name = mtx_run_accession + "-minus-" + mgx_run_accession
+            name = row['sample_name']
+            mtx_subtract_sigobj.name = name
+            
+            # create output sig file name
+            sig_filename = os.path.join('outputs/sourmash_sketch_subtract/' + name + "_k" + ksize + ".sig")
+            # write sig to file
+            with sourmash.sourmash_args.FileOutput(sig_filename, 'wt') as fp:
+                sourmash.save_signatures([mtx_subtract_sigobj], fp=fp)
+
```

Comment:
nit: extra newlines

General style advice is a singular newline at the EOF

Reply:
thanks! I added this when I was debugging an EOF syntax error and forgot to remove it!
```suggestion
```